File: src/main/kotlin/dev/jefy/connectpro/shared/application/dtos/PortfolioSummaryData.java
package dev.jefy.connectpro.shared.application.dtos;


import java.util.UUID;

import dev.jefy.connectpro.portfolio.domain.model.Portfolio;

/**
 * @author Jôph Yamba
 */
public record PortfolioSummaryData(
        UUID id,
        String name,
        String coverImageUrl,
        String shortDescription,
        boolean active
) {
    public static PortfolioSummaryData from(Portfolio portfolio) {
        String imageUrl = portfolio.getGeneralInfo().coverImageUrl() == null
                ? null
                : portfolio.getGeneralInfo().coverImageUrl().value();
        return new  PortfolioSummaryData(
                portfolio.getId().value(),
                portfolio.getGeneralInfo().name(),
                imageUrl,
                portfolio.getGeneralInfo().shortDescription(),
                portfolio.isActive()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/shared/application/dtos/PortfolioSummaryData.kt
package dev.jefy.connectpro.shared.application.dtos

import dev.jefy.connectpro.portfolio.domain.model.Portfolio
import dev.jefy.connectpro.shared.infrastructure.file_storage.ImageUrlResolver
import java.util.*

data class PortfolioSummaryData(
    val id: UUID,
    val name: String,
    val coverImage: String?,
    val shortDescription: String,
    val active: Boolean
) 

fun Portfolio.toSummaryData(resolver: ImageUrlResolver): PortfolioSummaryData {
    val image = this.generalInfo.coverImage
    return PortfolioSummaryData(
        id = this.id.value,
        name = this.generalInfo.name,
        coverImage = resolver.resolve(image),
        shortDescription = this.generalInfo.shortDescription,
        active = this.isActive()
    )
}